beacon_node/src/config.rs

```diff
@@ -222,16 +222,26 @@ pub fn get_configs<E: EthSpec>(
     match cli_args.subcommand() {
         ("testnet", Some(sub_cmd_args)) => {
             process_testnet_subcommand(&mut client_config, &mut eth2_config, sub_cmd_args)?
-        }
+        },
+        ("purge", _) => {
+            client_config.purge_chain_db()?;
+            println!("Successfully purged chain db");
+            std::process::exit(0);
```

Comment:
This seems a little bit strange, but I can also see how it seems like the best way given the current structure of the code. @paulhauner do you have any strong opinions on this?

Reply:
A cleaner design might be to return a `Result<NodeRunMode>`, where `NodeRunMode` is a new enum with a variant for running the beacon chain with some config, and another for doing a purge.

```rust
enum BeaconChainAction {
   Start(Config),
   Purge(Config),
}
```

Reply:
That doesn't appear to work from what I can see:

we are currently inside `get_configs()`
The call graph looks like this:
`run()->ProductionBeaconNode::new_from_cli()->get_configs()`

Correct me if I'm wrong, but I believe under ideal circumstances, we would shut down after purging by calling `environment.shutdown_on_idle() // as done in main.rs:209`

`environment` is only accessible from inside `run()`. I don't see a way to pass it to `ProductionValidatorClient::new_from_cli()`, so our only other option is to propagate the Purge state back up from 
**both** `get_configs()` **and** `ProductionValidatorClient::new_from_cli()`.

This could _almost_ be accomplished by making `NodeRunMode` generic:
```
enum NodeRunMode<T> {
    Start(T),
    Purge
}
```

But if we modify `new_from_cli()` to return a `Future< Item = NodeRunMode<Self>, Error = String>`, Rust will complain that `NodeRunMode` doesn't implement `futures::Future`

I'm reaching the limits of my n00b Rust knowledge here but I _think_ one option could be to put in a trait bound:
```
enum NodeRunMode<T: Future> {
    Start(T),
    Purge
}
impl Future for NodeRunMode<T> {
    // simple wrapper code that just calls the underlying type's implementation
}
```

But if you do this then Rust will complain about the return type from `get_configs` not implementing Future... you could get around this by implementing 2 NodeRunMode types but.. then I feel like this has gotten even more messy than calling `std::process::exit()`

Reply:
Ok, you're right. Let's stick with `std::process::exit` for now :+1: